identity.go

```diff
@@ -274,6 +390,28 @@ func (id *ID) GetConfigForClient(config *tls.Config, _ *tls.ClientHelloInfo) (*t
 	return config, nil
 }
 
+// queueReload de-duplicates reload attempts within a 5s window.
+func (id *ID) queueReload(closeNotify <-chan struct{}) {
+	id.needsReload.Store(true)
+	id.reloader.Do(func() {
+		go func() {
+			for {
+				select {
+				case <-time.After(1 * time.Second):
```

Comment:
Maybe instead of having a goroutine running all the time for reloads, just run this with a 1 second delay after a file reload has been detected with time.AfterFunc or similar. I'm assuming you're trying to avoid reloading once for each file changed when you could wait a second and reload them all together?

Reply:
Avoiding multiple reloads: Yes

If the files are being replaced through a tool it is likely all the files will change and only one reload is necessary.

Mmm sure can do that. Will give it a go.

Reply:
Fixed